graphistry/plugins/spannergraph.py

```diff
@@ -38,69 +40,55 @@ class SpannerGraph:
+        Initializes the SpannerGraph instance and establishes a connection to the Spanner database.
 
         :param graphistry: The Graphistry parent object.
         :param project_id: The Google Cloud project ID.
         :param instance_id: The Spanner instance ID.
         :param database_id: The Spanner database ID.
         """
         
-        # check if valid 
         required_keys = ["project_id", "instance_id", "database_id"]
         for key in required_keys:
             value = spanner_config.get(key)
-            if not value:  # check for None or empty values
+            if not value:
                 raise ValueError(f"Missing or invalid value for required Spanner configuration: '{key}'")
 
-        self.project_id = spanner_config["project_id"]
-        self.instance_id = spanner_config["instance_id"]
-        self.database_id = spanner_config["database_id"]
-    
-        if spanner_config.get("credentials_file"):
-            self.credentials_file = spanner_config["credentials_file"]
-   
-        self.connection = self.__connect()
-
-    def __connect(self) -> Any:
-        """
-        Establishes a connection to the Spanner database.
-
-        :return: A connection object to the Spanner database.
-        :rtype: google.cloud.spanner_dbapi.connection
-        :raises SpannerConnectionError: If the connection to Spanner fails.
-        """
-        from google.cloud.spanner_dbapi.connection import connect
+        project_id = spanner_config["project_id"]
+        instance_id = spanner_config["instance_id"]
+        database_id = spanner_config["database_id"]
+        credentials_file = spanner_config.get("credentials_file", None)
 
         try:
-            if hasattr(self, 'credentials_file') and self.credentials_file is not None:
-                
-                connection = connect(self.instance_id, self.database_id, credentials=self.credentials_file)
+            if credentials_file:
+                self.connection = connect(instance_id, database_id, credentials=credentials_file)
             else:
-                connection = connect(self.instance_id, self.database_id)
+                self.connection = connect(instance_id, database_id)
 
-            connection.autocommit = True
+            self.connection.autocommit = True # NOTE: Why would we set this?
             logger.info("Connected to Spanner database.")
-            return connection
+            return cls(self.connection)
+        
         except Exception as e:
             raise SpannerConnectionError(f"Failed to connect to Spanner: {e}")
 
-    def close_connection(self) -> None:
+
+    def close(self) -> None:
         """
         Closes the connection to the Spanner database.
         """
-        if self.connection:
-            self.connection.close()
-            logger.info("Connection to Spanner database closed.")
+        self.connection.close()
+        logger.info("Closed Spanner connection.")
```

Comment:
fwiw, does this mean the caller is supposed to do `res_g = g.kusto_query_to_graph(query); res_g._kusto_client.close()` ?

Reply:
(In general, unsure what lifecycle pattern we want for reconnecting, closing connections, etc, getting back to the whole louie discussion)